dra-evolution/pkg/api/capacity_types.go

```diff
@@ -64,12 +64,33 @@ type ResourcePoolSpec struct {
 	// Must not have more than 128 entries.
 	Devices []Device `json:"devices,omitempty"`
 
+	// SharedAllocatable are pooled resources that are shared by all
+	// devices in the pool. This is typically used when representing a
+	// partitionable device, and need not be populated otherwise.
+	//
+	// Must not have more than 32 entries.
+	//
+	// +optional
+	// +listType=atomic
+	SharedAllocatable []ResourceCapacity `json:"sharedAllocatable,omitempty"`
+
 	// FUTURE EXTENSION: some other kind of list, should we ever need it.
 	// Old clients seeing an empty Devices field can safely ignore the (to
 	// them) empty pool.
 }
 
+type ResourceCapacity struct {
+	// Name is the resource name/type.
+	// +required
+	Name string `json:"name"`
+
+	// Capacity is the total capacity of the named resource.
+	// +required
+	Capacity resource.Quantity `json:"capacity"`
```

Comment:
I'm still not sold on the name `Capacity` but I don't have anything better. I think I had used `Limit` at one point, but that's not any better.

So in yaml, this would look like:
```
apiVersion: resource.k8s.io/v1alpha3
kind: ResourcePool
metadata:
  name: node0-gpu0
spec:
  driver: gpu.nvidia.com
  nodeName: node0
  sharedConsumable:
  - name: memory
     capacity: 40Gi
  - name: multiprocessors
    capacity: "98"
  - name: copy-engines
    capacity: "7"
  - name: decoders
    capacity: "5"
  - name: encoders
    capacity: "0"
  - name: jpeg-engines
    capacity: "1"
  - name: ofa-engines
    capacity: "1"
  - name: memory-slices-0
    capacity: "1"
  - name: memory-slices-1
     capacity: "1"
  - name: memory-slices-2
    capacity: "1"
  - name: memory-slices-3
    capacity: "1"
  - name: memory-slices-4
    capacity: "1"
  - name: memory-slices-5
    capacity: "1"
  - name: memory-slices-6
    capacity: "1"
  - name: memory-slices-7
    capacity: "1"
```

Reply:
I guess that's not too terrible. although seeing this, I'm wondering if we can't just change `sharedConsumable` to `sharable`.

Reply:
"amount" seems ok too. But again, I am not too opinionated on the names.